jf_agent/agent_logging.py

```diff
@@ -217,6 +219,53 @@ def filter(self, record: LogRecord) -> bool:
         return not record.__dict__.get(AGENT_LOG_TAG, False)
 
 
+def configure_structlog() -> None:
+    """
+    Configures rendering of log messages within structlog, which then passes a fully formatted string to
+    the logging module for output.
+    """
+    structlog.stdlib.recreate_defaults()
+    structlog.configure(
+        processors=[
+            structlog.contextvars.merge_contextvars,
+            structlog.stdlib.filter_by_level,
+            structlog.stdlib.add_logger_name,
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            structlog.processors.TimeStamper(fmt="iso"),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.processors.UnicodeDecoder(),
+            structlog.processors.CallsiteParameterAdder(
+                {
+                    structlog.processors.CallsiteParameter.FILENAME,
+                    structlog.processors.CallsiteParameter.FUNC_NAME,
+                    structlog.processors.CallsiteParameter.LINENO,
+                }
+            ),
+            structlog.dev.ConsoleRenderer(),
+        ],
+        wrapper_class=structlog.stdlib.BoundLogger,
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        cache_logger_on_first_use=True,
+    )
+
+
+def bind_context_vars(
+    run_mode: str,
+    company_slug: Optional[str],
+    upload_time: str,
+) -> None:
+    structlog.contextvars.clear_contextvars()
+    structlog.contextvars.bind_contextvars(
+        run_mode=run_mode,
+        company_slug=company_slug,
+        upload_time=upload_time,
+        agent_run_uuid=str(uuid.uuid4()),
+        jf_meta={'commit': os.getenv("SHA"), 'timestamp': os.getenv("BUILDTIME")},
```

Comment:
Nit: `timestamp` isn't readily apparent as being the build time here; consider renaming to `commit_build_time` or something similar